### PR TITLE
[PyTorch][TF] Fix matrix-vector product text in linear-algebra

### LINKED ISSUE #1792: Inconsistent use of np.dot and torch.mv in Section 2.3.8
The final paragraph of Section 2.3.8 *Matrix-Vector Products* mentions the use of `np.dot` but not `torch.mv`. The subsequent code examples uses `torch.mv` but not `np.dot`.

Either the text should describe `torch.mv` or the code should use `np.dot`

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
Fixes #1792 

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1794/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1794/

**Comment by mli** [comment]
Job d2l-en/PR-1794/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1794/

File: chapter_preliminaries/linear-algebra.md [hunk 1]
```diff
@@ -674,8 +672,6 @@ tf.cumsum(A, axis=0)
 So far, we have only performed elementwise operations, sums, and averages. And if this was all we could do, linear algebra probably would not deserve its own section. However, one of the most fundamental operations is the dot product.
 Given two vectors $\mathbf{x}, \mathbf{y} \in \mathbb{R}^d$, their *dot product* $\mathbf{x}^\top \mathbf{y}$ (or $\langle \mathbf{x}, \mathbf{y}  \rangle$) is a sum over the products of the elements at the same position: $\mathbf{x}^\top \mathbf{y} = \sum_{i=1}^{d} x_i y_i$.
 
-[~~The *dot product* of two vectors is a sum over the products of the elements at the same position~~]
```
Comment (astonzhang) [inline chapter_preliminaries/linear-algebra.md:677]: why do we remove this?

File: chapter_preliminaries/linear-algebra.md [hunk 2]
```diff
@@ -777,12 +773,30 @@ to describe the most intensive calculations
...
 the matrix-vector product is performed.
 Note that the column dimension of `A` (its length along axis 1)
 must be the same as the dimension of `x` (its length).
+:end_tab:
+
+:begin_tab:`pytorch`
+Expressing matrix-vector products in code with tensors, we use
+the `mv` function. When we call `torch.mv(A, x)` with a matrix
+`A` and a vector `x`, the matrix-vector product is performed.
+Note that the column dimension of `A` (its length along dim 1)
```
Comment (astonzhang) [inline chapter_preliminaries/linear-algebra.md:789]: dim -> axis.
Besides, even using "dim", we can spell it out as "dimension" in proper English :)

File: chapter_preliminaries/linear-algebra.md [hunk 3]
```diff
@@ -777,12 +773,30 @@ to describe the most intensive calculations
...
+must be the same as the dimension of `x` (its length).
+:end_tab:
+
+:begin_tab:`tensorflow`
+Expressing matrix-vector products in code with tensors, we use
+the `matvec` function. When we call `tf.linalg.matvec(A, x)` with a
+matrix `A` and a vector `x`, the matrix-vector product is
+performed. Note that the column dimension of `A` (its length along dim 1)
```
Comment (astonzhang) [inline chapter_preliminaries/linear-algebra.md:797]: dim -> axis

**Comment by astonzhang** [comment]
LGTM except for comments above.

File: chapter_preliminaries/linear-algebra.md (hunk 1 above)
Comment (AnirudhDagar) [reply chapter_preliminaries/linear-algebra.md:677]: Ohh sorry, I forgot this was a part of the slides syntax. Thanks! I'll add this back.

File: chapter_preliminaries/linear-algebra.md (hunk 2 above)
Comment (AnirudhDagar) [reply chapter_preliminaries/linear-algebra.md]: Done!

**Pull request merged**